### ISSUE TITLE
Modal: Aria-label cannot be removed when not needed

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/vets-design-system-documentation#1407]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

The `<va-modal>` is automatically generating an `aria-label` from the `modalTitle `attribute. However, when `aria-labelledby` is added, `aria-label` is no longer needed and there is currently no way to entirely remove this. By adding an empty string into `modalTitle`, `aria-label` still displays as `aria-label=" modal"`

![image](https://user-images.githubusercontent.com/106678594/208435141-d0fe54f7-1f11-4240-affd-525164839e3e.png)

## What I expected to happen

- [ ] There is a way to completely remove `aria-label` when not needed

## Reproducing

1. Add a `<va-modal>` component
2. Add an empty string to the `modalTitle `attribute
3. Notice that the `aria-label` is not removed but instead it is generate as `aria-label=" modal"`

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

### Details
This was reported in slack and Robin assisted me with determining that this would need to be a bug report: https://dsva.slack.com/archives/C01DBGX4P45/p1671157719506679?thread_ts=1671127606.381169&cid=C01DBGX4P45 

When this is updated, it also might be helpful to update the [accessibility considerations](https://design.va.gov/components/modal#accessibility-considerations) to make sure engineers are aware of when to use the aria-label vs. the aria-labelledby and why both are not necessary.

**Comment by amyleadem** [comment]
@caw310 
cc @RyanMunsch 
This issue could be considered a duplicate of #2189, which requests a modal prop for the `aria-labelledby` attribute. From my understanding, PR https://github.com/department-of-veterans-affairs/component-library/pull/1646 should address the issue of removing the `aria-label` attribute when `aria-labelledby` is preferred. 

Flagging that we might be able to close this issue when https://github.com/department-of-veterans-affairs/component-library/pull/1646 is merged, depending on the outcome.

**Comment by caw310** [comment]
Closing as duplicate
